vllm/engine/arg_utils.py

```diff
@@ -20,6 +21,24 @@ def nullable_str(val: str):
     return val
 
 
+class FlexibleArgumentParser(argparse.ArgumentParser):
+    """ArgumentParser that allows both underscore and dash in names."""
+
+    def parse_args(self, args=None, namespace=None):
+        if args is None:
+            args = sys.argv[1:]
+
+        # Convert underscores to dashes in argument names
+        processed_args = []
+        for arg in args:
+            if arg.startswith('--'):
+                processed_args.append('--' + arg[2:].replace('_', '-'))
+            else:
+                processed_args.append(arg)
+
+        return super().parse_args(processed_args, namespace)
+
+
```

Comment:
this class is not specific to engine. can we move it to utils or have a separate file for it?

Reply:
Sure, I moved it to vllm/utils.py